gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java

```diff
@@ -448,14 +448,7 @@ public synchronized void requestTargetNumberOfContainers(YarnContainerRequestBun
       for(; requestedContainerCount < desiredContainerCount; requestedContainerCount++) {
         requestContainer(Optional.absent(), yarnContainerRequestBundle.getHelixTagResourceMap().get(currentHelixTag));
       }
-      requestedContainerCountMap.put(currentHelixTag, desiredContainerCount);
-    }
-
-    // If a requested tag is not presented in the new request, update the requested count to 0 as we should release them
-    for(String requestedHelixTag : requestedContainerCountMap.keySet()) {
-      if(!yarnContainerRequestBundle.getHelixTagContainerCountMap().containsKey(requestedHelixTag)) {
-        requestedContainerCountMap.put(requestedHelixTag, 0);
-      }
+      requestedContainerCountMap.put(currentHelixTag, requestedContainerCount);
```

Comment:
Inside this for loop we should only update(increase in this case) the map as we request new containers. Decreasing the map should be done only when we shrinking containers.